src/field/FieldPropagator.hh

```diff
@@ -0,0 +1,79 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file FieldPropagator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Macros.hh"
+#include "base/Types.hh"
+
+#include "field/FieldTrackView.hh"
+#include "field/FieldParamsPointers.hh"
+#include "field/FieldInterface.hh"
+#include "field/FieldDriver.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Brief class description.
+ *
+ *  This is a high level interface of propagating a charged particle in a
+ *  magnetic field.
+ *
+ * \note This follows similar methods as in Geant4's G4PropagatorInField class.
+ */
+class FieldPropagator
+{
+  public:
+    // Construct with shared parameters and the field driver
+    inline CELER_FUNCTION
+    FieldPropagator(const FieldParamsPointers& shared, FieldDriver& driver);
+
+    // Propagation in a field
+    inline CELER_FUNCTION real_type operator()(FieldTrackView* view);
```

Comment:
I think this highest-level class should look like the `LinearPropagator`, where the track view is part of the class constructor, and the operators take the physical step limit (or an option to move to the next geometry boundary with an "infinite" physical step).

Reply:
Contrary to the LinearPropagator, this operator also need information from ParticleTrackView (i.e, momemtum, charge, (global time, spin later)) which is why this intermediate FieldTrackView was originally introduced - the scope of its usage is now limited up to the high-level, but may be extended deeper later.   Again, the difference between LinearPropagator and the FieldPropagator is whether the (propagation) state (not necessarily the vecgeom navigatiion state for the field propagation) is occasionally updated or not.  I guess that it is up to the transport process what relevant information passes to the linear propagator for the neutral particle  and to the field propagator for the charged particle with a non-zero magnetic field.  Anyway, we can definitely discuss these infra-structure related consistency further and remove or move-round things later once the current implementation is efficiently working for the case with non-uniform magnetic field (which is the next target).